webapp/src/ts/modals/edit-user/update-password.component.ts

```diff
@@ -48,13 +53,27 @@ export class UpdatePasswordComponent extends EditUserAbstract implements OnInit
     this.errors = {};
     this.setProcessing();
     if (this.validatePasswordFields()) {
-      const updates = { password: this.editUserModel.password };
+      const password = this.editUserModel.password;
+      const updates = { password };
       const username = this.editUserModel.username;
       this.updateUserService
         .update(username, updates, username, this.editUserModel.currentPassword)
         .then(() => {
-          this.setFinished();
-          this.windowReload();
+          const data = this.getLoginData(username, password);
+          this.userLoginService
+            .login(data)
+            .catch(err => {
+              if (err.status === 302) {
+                this.setFinished();
+                this.close();
+
+                this.modalService
+                  .show(ConfirmPasswordUpdatedComponent)
+                  .catch(() => {});
```

Comment:
```suggestion
                  .catch(() => window.location.reload(true));
```

This solves the problem of clicking outside the confirmation dialog and the app not being reloaded.